datafusion/common/src/file_options/parquet_writer.rs

```diff
@@ -260,6 +289,11 @@ impl ParquetOptions {
             .set_data_page_row_count_limit(*data_page_row_count_limit)
             .set_bloom_filter_enabled(*bloom_filter_on_write);
 
+        if fep.is_some() {
+            builder = builder.with_file_encryption_properties(fep.unwrap());
+        }
```

Comment:
This part is key!! Essentially, I think all the writes go through the parquet writer so we can just integrate here.